tools/parse/src/main/scala/org/allenai/nlpstack/parse/poly/core/SentenceTagger.scala

```diff
@@ -0,0 +1,279 @@
+package org.allenai.nlpstack.parse.poly.core
+
+import java.io.File
+
+import com.typesafe.config.{ Config, ConfigFactory }
+import org.allenai.common.Config.EnhancedConfig
+import org.allenai.nlpstack.core.{ Lemmatized, PostaggedToken, Postagger }
+import org.allenai.nlpstack.lemmatize.MorphaStemmer
+import org.allenai.nlpstack.parse.poly.ml._
+import org.allenai.nlpstack.parse.poly.polytagger.SimplePostagger
+import org.allenai.nlpstack.postag._
+import reming.DefaultJsonProtocol._
+
+import scala.concurrent.{ Await, Future }
+import scala.concurrent.duration._
+import scala.language.postfixOps
+
+/** Recipe for initializing a sentence tagger.
+  *
+  * When providing a new implementation of the SentenceTagger interface, please provide an
+  * associated SentenceTaggerInitializer here so that it can be used.
+  */
+sealed trait SentenceTaggerInitializer
+
+/** Initializes a default Factorie part-of-speech tagger.
+  *
+  * @param useCoarseTags set to true if you want the tagger to output Google coarse POS tags
+  */
+case class FactoriePostaggerInitializer(useCoarseTags: Boolean) extends SentenceTaggerInitializer
```

Comment:
I know you do this a lot, but mixing configuration parameters into the type system makes me sad.